### PR TITLE
Link subsequent data world coords to only the first compatible

### PR DESCRIPTION
**Pull request by rosteen**
Thanks to @astrofrog for pointing out that we don't need to link every dataset to every other dataset, we can just link all subsequent data to the first. This change makes the performance scale linearly instead of exponentially as more datasets are added, which vastly improves the behavior of #394 and also helps in cases of loading large quantities of data into Specviz. We'll likely want to do some follow up work to further restrict when things get linked, since I don't think we need to link the spectra at all in the Mosviz case.

### CODE REVIEW

**Comment by rosteen** [comment]
@camipacifici I tagged you in as a reviewer in the hopes that you would try this with the case you'd mentioned before of loading lots of data into Specviz.

**Comment by rosteen** [comment]
I have a few test fixes to do, need to check for less links since, well, there are less links now.

File: jdaviz/app.py
Comment (pllim) [inline jdaviz/app.py:249]: I can't say I understand that is exactly happening but I am curious: Is this the 3-minute solution or a faster one that @astrofrog suggested?

Comment (rosteen) [reply jdaviz/app.py:249]: This is the low-effort solution that knocks the NIRISS loading down to 3 minutes. I figured we could get this in as a quick significant improvement and build off of it to make further performance gains later.

**Review by pllim — APPROVED** [review]
LGTM in theory, as long as we agree that this should be further improved in a future PR.

You might want to wait till Cami etc can confirm speed up in practice before merge.

**Review by javerbukh — APPROVED** [review]
I can confirm that this improves speed of data loading significantly. I can rebase #394 once this is merged to make sure it works as expected on NIRISS data. Nice work!

**Review by camipacifici — APPROVED** [review]
This is great! Thank you for hunting this down!
Approved!

**Comment by javerbukh** [comment]
Merging.

**Pull request merged**
